### ISSUE TITLE
Warn about unknown manifest keys

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-linter#1074]
### Describe the problem and steps to reproduce it:

* Create an extension with manifest that has an unsupported key, such as `comands` instead of [commands](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/commands).
* Run the linter

### What happened?

It should warn you that the key is unknown.

### What did you expect to happen?

Nothing happened. The linter reported no errors, warnings, or notices.

### Anything else we should know?

Chrome warns you about it like this in the extension panel: 
![screenshot 2016-12-14 15 47 02](https://cloud.githubusercontent.com/assets/55398/21202671/ee5d93ac-c214-11e6-903b-d32cd3b7697d.png)

**Comment by andymckay** [comment]
Think this should depend upon #765

**Comment by EnTeQuAk** [comment]
cc @mstriemer on this to find out whether or not this is already planned or easily implementable.

**Comment by mstriemer** [comment]
This doesn't happen by default with the new schemas. We could add `additionalProperties: false` to the manifest but then it appears as an error.

I think we'd want to update the `errorLookup()` function parsers/manifestjson.js to downgrade that to a warning if `error.type === 'additionalProperties' && error.dataPath.lastIndexOf('/') === 0`. So it's an additionalProperties error at `/foo`.

**Comment by andymckay** [comment]
Note Firefox gives us this:

<img width="923" alt="screenshot 2017-08-11 17 36 26" src="https://user-images.githubusercontent.com/74699/29236153-a490dd7e-7ebb-11e7-8870-0d2610d17693.png">

**Comment by Divya063** [comment]
@mstriemer Can I take this issue?

**Comment by EnTeQuAk** [comment]
@Divya063 go for it! It's open for contributions. Please let us know if you run into any trouble or need any assistance

**Comment by Divya063** [comment]
@EnTeQuAk  Should I create a separate error message like MANIFEST_INVALID_KEY  in messages/manifestjson.js or just go with  `error.message = 'is not a valid key or has invalid extra properties' ` ?

**Comment by EnTeQuAk** [comment]
Yes, please add a new error message in `messages/manifestjson.js` and maybe take a look at how `PROP_NAME_MISSING` and it's respective `manifestPropMissing` work on how to inject a property key into the error message.

So something along the way of `An unexpected property "unknownkey" was found in the WebExtension manifest.` might work.

**Comment by Divya063** [comment]
@EnTeQuAk Thanks for the suggestion!

**Comment by Divya063** [comment]
I'm getting this error 
` Cannot read property 'defineRule' of undefined
(node:11812) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'defineRule' of undefined
`

**Comment by EnTeQuAk** [comment]
@Divya063 as mentioned in another issue, did you run `npm install` in your local checkout to update all dependencies? This error should have been fixed in the most recent eslint upgrade.

**Comment by Divya063** [comment]
Hi @EnTeQuAk,  I'm quite confused about creating a new error message as there is MANIFEST_FIELD_INVALID already present which I guess is for the same purpose (ie creating warning about  invalid keys)

**Comment by mstriemer** [comment]
I think that error is used for invalid values in the manifest. It would likely be best to add a new error called `MANIFEST_FIELD_UNKNOWN`. You'll want to add it to `src/messages/manifestjson.js` and `docs/rules.md` in a similar fashion to `MANIFEST_FIELD_INVALID`.

We expect `MANIFEST_FIELD_INVALID` to be an error and I'm guessing we want this to be a warning in case this is a cross browser extension and we don't support that manifest field.

**Comment by laedit** [comment]
Sorry to dig into this, but after some test in Firefox 56 and 59.0a1 and with the following manifest:

``` json
{
  "manifest_version": 2,
  "name": "webext-test",
  "version": "0.1",
  "description": "webext-test description",
  "permissions": [
    "storage"
  ],
  "browser_action": {
    "browser_style": true,
    "default_title": "webext-test(browserAction)"
  },
  "applications": {
    "gecko": {
      "id": "webext-test@laedit"
    },
    "engines": {
      "firefox": ">=38.0a1"
    }
  }
}
```
(`engines` key have been migrated from an old addon), it seems that Firefox doesn't load the extension at all and display this ![ff_invalid_extension](https://user-images.githubusercontent.com/871092/33892890-490cc936-df5a-11e7-91b5-6a7c7a079b01.png)

So I think that the linter should display it as an error and not a warning.

**Comment by EnTeQuAk** [comment]
Thanks @laedit for the update on this (and sorry for the late reply). I do agree that this should very well be an error, especially if Firefox behaves that way too.

**Comment by rpl** [comment]
@EnTeQuAk is this still an issue or is it something we already fixed? would you mind to double-check it?
